### PR TITLE
build: Update styled-components - [FE-4054]

### PR DESCRIPTION
**Pull request by dawidzarzycki**
### Proposed behaviour
**SCOPE OF CHANGES:**
- update dependencies 
  -  "styled-components": "^5.1.0"
  - "jest-styled-components": "^7.0.4"
- update tests
    - update snapshots
    - mising semicolons
    - fix tests
    
### Current behaviour
- "styled-components": "^4.4.1"
- "jest-styled-components": "^6.3.1"

### Checklist

- [x] Commits follow our style guide
- [ ] <del> Screenshots are included in the PR if useful
- [ ] <del> All themes are supported if required
- [x] Unit tests added or updated if required
- [ ] <del> Cypress automation tests added or updated if required
- [ ] <del> Storybook added or updated if required
- [ ] <del> Typescript `d.ts` file added or updated if required
- [ ] <del> Carbon implementation and Design System documentation are congruent

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -122,8 +122,8 @@
     "react-test-renderer": "^16.12.0",
     "rimraf": "^3.0.2",
     "semantic-release": "^17.4.0",
+    "styled-components": "^5.3.0",
```
Comment (mkrds) [inline package.json:125]: Why 2 different versions of `styled-components` ?

Comment (dawidzarzycki) [reply package.json:125]: Oh good point! I'll fix it

Comment (dawidzarzycki) [reply package.json:125]: @mkrds fixed 👍

File: package-lock.json [hunk 2]
```diff
@@ -10529,22 +10529,14 @@
...
+      "version": "3.0.0",
+      "resolved": "https://registry.npmjs.org/css-to-react-native/-/css-to-react-native-3.0.0.tgz",
+      "integrity": "sha512-Ro1yETZA813eoyUp2GDBhG2j+YggidUmzO1/v9eYBKR2EHVEniE2MI/NqpTQ954BMpTPZFsGNPm46qFB9dpaPQ==",
       "dev": true,
       "requires": {
         "camelize": "^1.0.0",
         "css-color-keywords": "^1.0.0",
-        "postcss-value-parser": "^3.3.0"
-      },
-      "dependencies": {
```
Comment (ianforster-sage) [inline package-lock.json:10541]: @sjohnson-sage what impact does this removal have on this open dependabot alert? https://github.com/Sage/carbon/security/dependabot/package-lock.json/postcss/open

File: src/__experimental__/components/date-range/__snapshots__/date-range.spec.js.snap [hunk 3]
```diff
@@ -1,93 +1,13 @@
...
 exports[`StyledDateRange renders Date inputs correctly when the labels are inline 1`] = `
-.c2 .c13 {
+.c2 .sc-bqGGPW {
   -webkit-flex: none;
   -ms-flex: none;
   flex: none;
   width: 135px;
 }
 
-.c3 .c1 {
```
Comment (ianforster-sage) [inline src/__experimental__/components/date-range/__snapshots__/date-range.spec.js.snap:11]: Why is there so much deletion from this snapshot?

File: src/__experimental__/components/radio-button/__snapshots__/radio-button.spec.js.snap [hunk 4]
```diff
@@ -56,269 +41,30 @@ exports[`RadioButton base renders as expected 1`] = `
...
 }
 
-.c2 .c17 {
+.c3 .sc-bdnxRM {
   -webkit-flex-basis: 100%;
   -ms-flex-preferred-size: 100%;
   flex-basis: 100%;
 }
 
-.c13 .c5 {
```
Comment (ianforster-sage) [inline src/__experimental__/components/radio-button/__snapshots__/radio-button.spec.js.snap:78]: Another snapshot with lots of removed styles. What's the story behind this?

File: src/__experimental__/components/switch/__snapshots__/switch-slider.spec.js.snap [hunk 5]
```diff
@@ -52,45 +52,17 @@ exports[`SwitchSlider base theme renders as expected 1`] = `
...
   right: -30px;
   height: 100%;
 }
 
-.c3 .c1 {
-  color: rgba(0,0,0,0.55);
-}
-
-.c4 .c1 {
-  color: rgba(0,0,0,0.55);
```
Comment (ianforster-sage) [inline src/__experimental__/components/switch/__snapshots__/switch-slider.spec.js.snap:66]: More deleted styles here

File: src/__experimental__/components/switch/__snapshots__/switch.spec.js.snap [hunk 6]
```diff
@@ -1,171 +1,75 @@
...
 }
 
-.c1 .c38 {
+.c1 .sc-bdnxRM {
   -webkit-flex-basis: 100%;
   -ms-flex-preferred-size: 100%;
   flex-basis: 100%;
 }
 
-.c19 .c4 {
```
Comment (ianforster-sage) [inline src/__experimental__/components/switch/__snapshots__/switch.spec.js.snap:80]: more removed styles

File: src/components/button-toggle/__snapshots__/button-toggle.spec.js.snap [hunk 7]
```diff
@@ -44,23 +44,11 @@ input:focus ~ .c2 {
...
 }
 
-.c2:hover {
+.c3:hover {
   background-color: #E6F5E6;
   border-color: #004500;
 }
 
-.c5 .c3 {
-  -webkit-flex-direction: column;
```
Comment (ianforster-sage) [inline src/components/button-toggle/__snapshots__/button-toggle.spec.js.snap:53]: And here

File: src/components/split-button/__snapshots__/split-button.spec.js.snap [hunk 8]
```diff
@@ -74,122 +74,22 @@ exports[`SplitButton for business themes renders styles correctly 1`] = `
...
   margin-bottom: 0px;
   height: 16px;
 }
 
-.c2 .c4 svg {
+.c2 .c6 svg {
   margin-top: 0;
 }
 
-.c6:hover .c4 {
```
Comment (ianforster-sage) [inline src/components/split-button/__snapshots__/split-button.spec.js.snap:92]: ditto

File: src/components/table/table-row/__snapshots__/table-row.spec.js.snap [hunk 9]
```diff
@@ -89,242 +89,6 @@ exports[`TableRow render if is not classic theme renders a row to match the snap
   border-right: 1px solid transparent;
 }
 
-.c6 .c3 {
-  background-color: #FFFFFF;
```
Comment (ianforster-sage) [inline src/components/table/table-row/__snapshots__/table-row.spec.js.snap:93]: ditto

File: src/components/tile/__snapshots__/tile.spec.js.snap [hunk 10]
```diff
@@ -21,26 +21,11 @@ exports[`Tile renders base styles 1`] = `
...
 }
 
-.c2 + .c1 {
+.c1 + .c1 {
   margin-top: 0;
   border-left: solid 1px #E6EBED;
 }
 
-.c7 + .c1 {
-  margin-top: 0;
```
Comment (ianforster-sage) [inline src/components/tile/__snapshots__/tile.spec.js.snap:30]: ditto

File: package-lock.json (hunk 2 above)
Comment (sjohnson-sage) [reply package-lock.json:10541]: None unfortunately!

**Comment by samtjo** [comment]
This work is no longer required, closing the PR

**Pull request closed without merging**